src/test/ui/span/coerce-suggestions.stderr

```diff
@@ -0,0 +1,52 @@
+error[E0308]: mismatched types
+  --> $DIR/coerce-suggestions.rs:17:20
+   |
+17 |     let x: usize = String::new();
+   |                    ^^^^^^^^^^^^^ expected usize, found struct `std::string::String`
+   |
+   = note: expected type `usize`
+   = note:    found type `std::string::String`
+   = help: here are some functions which might fulfill your needs:
+ - .capacity()
+ - .len()
+
+error[E0308]: mismatched types
+  --> $DIR/coerce-suggestions.rs:23:19
+   |
+23 |     let x: &str = String::new();
+   |                   ^^^^^^^^^^^^^ expected &str, found struct `std::string::String`
+   |
+   = note: expected type `&str`
+   = note:    found type `std::string::String`
+   = help: try with `&String::new()`
+
+error[E0308]: mismatched types
+  --> $DIR/coerce-suggestions.rs:30:10
+   |
+30 |     test(&y);
+   |          ^^ types differ in mutability
+   |
+   = note: expected type `&mut std::string::String`
+   = note:    found type `&std::string::String`
+   = help: try with `&mut y`
```

Comment:
I couldn't test since the current code was wrong. Once updated, this will follow.